### PR TITLE
scarab: init at 1.19.0.0

### PR DESCRIPTION
**Pull request by huantianad**
###### Description of changes

[Scarab](https://github.com/fifty-six/Scarab) is a Hollow Knight mod installer and manager. Tested basic mod installation and removal, enable/disable, and file selector.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/scarab/default.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    libSM
+    fontconfig
+    gtk3
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    graphicsmagick
+    wrapGAppsHook
+    gtk3
```
Comment (IvarWithoutBones) [inline pkgs/games/scarab/default.nix:49]: This should be in `buildInputs` instead, as no binaries from it get executed during the build.

File: pkgs/games/scarab/default.nix [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  ];
+
+  postFixup = ''
+    # Icon for .desktop
```
Comment (IvarWithoutBones) [inline pkgs/games/scarab/default.nix:53]:
```suggestion
    # Icon for the desktop file
```

File: pkgs/games/scarab/default.nix [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    install -d -m755 $out/share/icons/hicolor/256x256/apps/
```
Comment (IvarWithoutBones) [inline pkgs/games/scarab/default.nix:54]: Shouldn't this be an `mkdir` instead? The icon probably doesn't need these permissions.

File: pkgs/games/scarab/default.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "10pmzy7nhcam0686kpn64cgar59shzzy5k7j3vrgmnm27zgkr22v";
+  };
+
+  nugetDeps = ./deps.nix;
+  projectFile = "Scarab.sln";
+  executables = [ "Scarab" ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_6_0;
```
Comment (IvarWithoutBones) [inline pkgs/games/scarab/default.nix:33]: This is the default.
```suggestion
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
This should also get moved to `pkgs/tools/games`.

**Comment by huantianad** [comment]
@IvarWithoutBones Is there a specific place in `all-packages.nix` that packages in `pkgs/tools/games` should be placed? they seem to be just randomly scattered around.

File: pkgs/games/scarab/default.nix (hunk 3 above)
Comment (huantianad) [reply pkgs/games/scarab/default.nix:54]: Yeah I forgot that those permissions should be the default for `mkdir`.

**Review by IvarWithoutBones — APPROVED** [review]
Thanks for the PR! This looks good to me :)

Result of `nixpkgs-review pr 184413` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scarab</li>
  </ul>
</details>

**Comment by huantianad** [comment]
This should close #184413, forgot about that.

**Pull request merged**